### PR TITLE
APPS-1232 nextflow basic

### PR DESCRIPTION
**Pull request by mhrvol**

### CODE REVIEW

File: src/python/dxpy/templating/templates/nextflow/nextflow.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    echo "=== NF work-dir : ${DX_WORK}"
+    echo "=== NF Resume ID: ${NXF_UUID}"
+    echo "=== NF log file : ${DX_LOG}"
+    echo "=== NF cache    : $DX_PROJECT_CONTEXT_ID:/.nextflow/cache/$NXF_UUID"
+    echo "============================================================="
+
+    filtered_inputs=""
+    
+    @@RUN_INPUTS@@
+    nextflow -trace nextflow.plugin $nf_run_args_and_pipeline_params -log ${LOG_NAME} run /home/dnanexus/nfp @@PROFILE_ARG@@ -name ${NXF_UUID} $nf_run_args_and_pipeline_params ${filtered_inputs}
```
Comment (YuxinShi0423) [inline src/python/dxpy/templating/templates/nextflow/nextflow.sh:70]:
```suggestion
    nextflow -trace nextflow.plugin $nf_run_args_and_pipeline_params -log ${LOG_NAME} run /home/dnanexus/nfp @@PROFILE_ARG@@ -name $test-{NXF_UUID} $nf_run_args_and_pipeline_params ${filtered_inputs}
```
https://staging.dnanexus.com/projects/GG1P1p80728ypkgp9vyYg2kX/monitor/job/GG2byzQ0728QkYvY78bGfBjB
ran into error saying 

> Not a valid run name: `76f125c8-637e-4696-a51f-ddcf8d72b8ce` -- It must match the pattern ^[a-z](?:[a-z\d]|[-_](?=[a-z\d])){0,79}$

Comment (YuxinShi0423) [inline src/python/dxpy/templating/templates/nextflow/nextflow.sh]: Why we are given `$nf_run_args_and_pipeline_params` twice? @mhrvol Shall the first be `nf_advanced_opts`?

File: src/python/dxpy/templating/templates/nextflow/nextflow.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    echo "=== NF work-dir : ${DX_WORK}"
+    echo "=== NF Resume ID: ${NXF_UUID}"
+    echo "=== NF log file : ${DX_LOG}"
+    echo "=== NF cache    : $DX_PROJECT_CONTEXT_ID:/.nextflow/cache/$NXF_UUID"
+    echo "============================================================="
+
+    filtered_inputs=""
+    
+    @@RUN_INPUTS@@
+    nextflow -trace nextflow.plugin $nf_run_args_and_pipeline_params -log ${LOG_NAME} run /home/dnanexus/nfp @@PROFILE_ARG@@ -name $test-{NXF_UUID} $nf_run_args_and_pipeline_params ${filtered_inputs}
```
Comment (YuxinShi0423) [inline src/python/dxpy/templating/templates/nextflow/nextflow.sh:70]:
```suggestion
    nextflow -trace nextflow.plugin $nf_advanced_opts -log ${LOG_NAME} run /home/dnanexus/nfp @@PROFILE_ARG@@ -name $test-{NXF_UUID} $nf_run_args_and_pipeline_params ${filtered_inputs}
```

File: src/python/dxpy/templating/templates/nextflow/nextflow.sh (hunk 1 above)
Comment (mhrvol) [reply src/python/dxpy/templating/templates/nextflow/nextflow.sh]: Yes, Thanks!

File: src/python/dxpy/templating/templates/nextflow/nextflow.sh [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    echo "=== NF work-dir : ${DX_WORK}"
+    echo "=== NF Resume ID: ${NXF_UUID}"
+    echo "=== NF log file : ${DX_LOG}"
+    echo "=== NF cache    : $DX_PROJECT_CONTEXT_ID:/.nextflow/cache/$NXF_UUID"
+    echo "============================================================="
+
+    filtered_inputs=""
+    
+    @@RUN_INPUTS@@
+    nextflow -trace nextflow.plugin $nf_advanced_opts -log ${LOG_NAME} run /home/dnanexus/nfp @@PROFILE_ARG@@ -name test-${NXF_UUID} $nf_run_args_and_pipeline_params ${filtered_inputs}
```
Comment (YuxinShi0423) [inline src/python/dxpy/templating/templates/nextflow/nextflow.sh:70]:
```suggestion
    mkdir -p .nextflow/cache/$NXF_UUID

    nextflow -trace nextflow.plugin $nf_advanced_opts -log ${LOG_NAME} run /home/dnanexus/nfp @@PROFILE_ARG@@ -name test-${NXF_UUID} $nf_run_args_and_pipeline_params ${filtered_inputs}
```

File: src/python/dxpy/templating/templates/nextflow/nextflow.sh [hunk 4]
```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+on_exit() {
+  ret=$?
+  # upload log file
+  dx upload $LOG_NAME --path $DX_LOG --wait --brief --no-progress --parents || true
+  # backup cache
+  echo "=== Execution complete — uploading Nextflow cache metadata files"
+  dx rm -r "$DX_PROJECT_CONTEXT_ID:/.nextflow/cache/$NXF_UUID/*" 2>&1 >/dev/null || true
+  dx upload "/.nextflow/cache/$NXF_UUID" --path "$DX_PROJECT_CONTEXT_ID:/.nextflow/cache/$NXF_UUID" --no-progress --brief --wait -p -r || true
```
Comment (YuxinShi0423) [inline src/python/dxpy/templating/templates/nextflow/nextflow.sh:10]:
```suggestion
  dx upload ".nextflow/cache/$NXF_UUID" --path "$DX_PROJECT_CONTEXT_ID:/.nextflow/cache/$NXF_UUID" --no-progress --brief --wait -p -r || true
```

File: src/python/dxpy/templating/templates/nextflow/dxapp.json [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  "runSpec": {
+    "interpreter": "bash",
+    "execDepends": [
+      {
+        "name": "default-jre"
+      }
+    ],
+    "distribution": "Ubuntu",
+    "release": "20.04",
+    "file": "nextflow.sh",
```
Comment (YuxinShi0423) [inline src/python/dxpy/templating/templates/nextflow/dxapp.json:69]: minor: put this under `src/`?

**Pull request closed without merging**